### DISCUSSION TITLE
Things a bignner should know before submiting PRs.

### DISCUSSION THREAD

**Discussion by sashankthapa** [meshery/meshery#9346] [Q&A] [answered]
What are the things I should know before submitting PRs and any tips or mistakes to avoid?? thank you for your assist.

**Comment by saurabh100ni** [comment, marked as answer]
Hey @kozuedoingregression, 

Certainly! Here are some key considerations to keep in mind when raising a Pull Request (PR):

### **1. Ask For an Assignment on the Issue Before Raising a PR:**
Before initiating a PR, it is advisable to request an assignment on the relevant issue. This helps in avoiding potential conflicts where two contributors might submit PRs for the same issue simultaneously, preventing ambiguity in the process.

### **2. Sign Your Commits:**
Ensure that you sign your commits to maintain accountability and traceability. You can achieve this by using the `-s` flag while executing the commit command. For instance:
`git commit -s -m "Implementing feature XYZ"`

### **3. Join Our Meetings To Show Your Work:**
Following the PR submission, it is encouraged to showcase your work during our meetings to gather constructive feedback. You can find information about meeting schedules and details in our [community calendar](https://layer5.io/community/calendar).

For detailed info, please refer to [General Contribution Flow](https://layer5.io/community/handbook/contribution).
